src/main/java/seedu/address/ui/CalendarView.java

```diff
@@ -0,0 +1,227 @@
+package seedu.address.ui;
+
+import java.time.DayOfWeek;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Comparator;
+import java.util.logging.Logger;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ListChangeListener;
+import javafx.collections.ObservableList;
+import javafx.fxml.FXML;
+import javafx.scene.control.Label;
+import javafx.scene.control.ListCell;
+import javafx.scene.control.ListView;
+import javafx.scene.layout.Region;
+import seedu.address.commons.core.LogsCenter;
+import seedu.address.logic.Logic;
+import seedu.address.model.session.RecurringSession;
+import seedu.address.model.session.Session;
+import seedu.address.model.session.SessionDate;
+import seedu.address.model.student.Student;
+import seedu.address.model.tuition.Tuition;
+
+public class CalendarView extends UiPart<Region> {
+    private static final Logger logger = LogsCenter.getLogger(CalendarView.class);
+    private static final String FXML = "CalendarView.fxml";
+
+    private Logic logic;
+    private ObservableList<Tuition> tuitionList;
+
+    // must be a monday, change through setStartDate()
+    private LocalDateTime calendarStartDate;
+
+    // FXML date headers (mon - sun)
+    @FXML private Label dateHeaderMon;
+    @FXML private Label dateHeaderTue;
+    @FXML private Label dateHeaderWed;
+    @FXML private Label dateHeaderThur;
+    @FXML private Label dateHeaderFri;
+    @FXML private Label dateHeaderSat;
+    @FXML private Label dateHeaderSun;
+
+
+    // FXML list views (mon - sun)
+    @FXML private Label dateLabel;
+    @FXML private ListView<Tuition> monListView;
+    @FXML private ListView<Tuition> tueListView;
+    @FXML private ListView<Tuition> wedListView;
+    @FXML private ListView<Tuition> thurListView;
+    @FXML private ListView<Tuition> friListView;
+    @FXML private ListView<Tuition> satListView;
+    @FXML private ListView<Tuition> sunListView;
+
+    /**
+     * Creates a {@code CalendarView} with the given {@code Logic}.
+     */
+    public CalendarView(Logic logic) {
+        super(FXML);
+        this.logic = logic;
+
+        tuitionList = FXCollections.observableArrayList();
+        LocalDateTime currentMonday = LocalDateTime.now().with(DayOfWeek.MONDAY);
+
+        setStartDate(currentMonday);
+        populateTuitions();
+        setStudentListListener();
+        setListViews(tuitionList);
+    }
+
+    class CalendarViewCell extends ListCell<Tuition> {
+        @Override
+        protected void updateItem(Tuition tuition, boolean empty) {
+            super.updateItem(tuition, empty);
+
+            if (empty || tuition == null) {
+                setGraphic(null);
+                setText(null);
+            } else {
+                setGraphic(new CalendarCard(tuition).getRoot());
+            }
+        }
+    }
+
+    /**
+     * Changes the start date of the week displayed on calendar and sessions associated with the week
```

Comment:
Actually abit OCD, but as per the java coding standard for javaDocs, maybe adding a full stop to all of the javaDoc comments might be more consistent.

Reply:
ok got it! will make the changes